Zilon.Core/Zilon.Core/PersonModules/HumanSurvivalModule.cs

```diff
@@ -208,5 +289,261 @@ public override void ResetStats()
+                        // Но case- блок должен быть, иначе будет ошибка.
+                        break;
+
+                    default:
+                        throw new InvalidOperationException($"Неизвестный тип эффекта {effect.GetType()}");
+                }
+            }
+        }
+
+        private void FillSurvivalBonusesFromPerks([NotNull, ItemNotNull] ref List<SurvivalStatBonus> bonusList)
+        {
+            if (_evolutionModule is null)
+            {
+                return;
+            }
+
+            var archievedPerks = _evolutionModule.GetArchievedPerks();
+            foreach (var archievedPerk in archievedPerks)
+            {
+                var currentLevel = archievedPerk.CurrentLevel;
+                var currentLevelScheme = archievedPerk.Scheme.Levels[currentLevel.Primary];
+
+                if (currentLevelScheme.Rules == null)
+                {
+                    continue;
+                }
+
+                for (var i = 0; i <= currentLevel.Sub; i++)
+                {
+                    foreach (var rule in currentLevelScheme.Rules)
+                    {
+                        if (rule.Type == PersonRuleType.Health)
+                        {
+                            BonusToHealth(rule.Level, PersonRuleDirection.Positive, ref bonusList);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void ApplySurvivalBonuses(IEnumerable<SurvivalStatBonus> bonuses)
+        {
+            foreach (var bonus in bonuses)
+            {
+                var stat = Stats.SingleOrDefault(x => x.Type == bonus.SurvivalStatType);
+                if (stat != null)
+                {
+                    var normalizedValueBonus = (int)Math.Round(bonus.ValueBonus, MidpointRounding.AwayFromZero);
+                    stat.ChangeStatRange(stat.Range.Min, stat.Range.Max + normalizedValueBonus);
+                    var normalizedDropPassBonus = (int)Math.Round(bonus.DownPassBonus, MidpointRounding.AwayFromZero);
+                    stat.DownPassRoll = SurvivalStat.DEFAULT_DOWN_PASS_VALUE - normalizedDropPassBonus;
+                }
+            }
+        }
+
+        private void FillSurvivalBonusesFromEquipments([NotNull, ItemNotNull] ref List<SurvivalStatBonus> bonusList)
+        {
+            if (_equipmentModule is null)
+            {
+                return;
+            }
+
+            var equipmentModule = _equipmentModule;
+
+            for (var i = 0; i < equipmentModule.Count(); i++)
+            {
+                var equipment = equipmentModule[i];
+                if (equipment == null)
+                {
+                    continue;
+                }
+
+                var rules = equipment.Scheme.Equip.Rules;
+
+                if (rules == null)
+                {
+                    continue;
+                }
+
+                foreach (var rule in rules)
+                {
+                    switch (rule.Type)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5480188)